resource-managers/kubernetes/core/src/test/scala/org/apache/spark/scheduler/cluster/kubernetes/ExecutorPodFactorySuite.scala

```diff
@@ -0,0 +1,228 @@
+      NodeAffinityExecutorPodModifierImpl, None, None, None, None, Some(shuffleManager))
+    val executor = factory.createExecutorPod("1",
+      "dummy", "dummy", Seq[(String, String)](), driverPod, Map[String, Int]())
+
+
+    assert(executor.getSpec.getContainers.size() == 1)
+    assert(executor.getSpec.getContainers.get(0).getVolumeMounts.size() == 1)
+    assert(executor.getSpec.getContainers.get(0).getVolumeMounts.get(0).getName == "0-tmp")
+    assert(executor.getSpec.getContainers.get(0).getVolumeMounts.get(0).
+      getMountPath == "/tmp")
+    checkOwnerReferences(executor, driverPodUid)
+  }
+
+  test("Small-files add a secret & secret volume mount to the container") {
+    val conf = baseConf.clone()
+    val smallFiles = new MountSmallFilesBootstrapImpl("secret1", "/var/secret1")
+
+    val factory = new ExecutorPodFactoryImpl(conf,
+      NodeAffinityExecutorPodModifierImpl, None, Some(smallFiles), None, None, None)
+    val executor = factory.createExecutorPod("1",
+      "dummy", "dummy", Seq[(String, String)](), driverPod, Map[String, Int]())
+
+
+    assert(executor.getSpec.getContainers.size() == 1)
+    assert(executor.getSpec.getContainers.get(0).getVolumeMounts.size() == 1)
+    assert(executor.getSpec.getContainers.get(0).getVolumeMounts.get(0)
+      .getName == "submitted-files")
+    assert(executor.getSpec.getContainers.get(0).getVolumeMounts.get(0)
+      .getMountPath == "/var/secret1")
+
+    assert(executor.getSpec.getVolumes.size() == 1)
+    assert(executor.getSpec.getVolumes.get(0).getSecret.getSecretName == "secret1")
+
+    checkOwnerReferences(executor, driverPodUid)
+    checkEnv(executor, Set("SPARK_MOUNTED_FILES_FROM_SECRET_DIR"))
+  }
+
+  test("classpath and extra java options get translated into environment variables") {
+    val conf = baseConf.clone()
+    conf.set(org.apache.spark.internal.config.EXECUTOR_JAVA_OPTIONS, "foo=bar")
+    conf.set(org.apache.spark.internal.config.EXECUTOR_CLASS_PATH, "bar=baz")
+
+    val factory = new ExecutorPodFactoryImpl(conf,
+      NodeAffinityExecutorPodModifierImpl, None, None, None, None, None)
+    val executor = factory.createExecutorPod("1",
+      "dummy", "dummy", Seq[(String, String)]("qux" -> "quux"), driverPod, Map[String, Int]())
+
+    checkEnv(executor, Set("SPARK_JAVA_OPT_0", "SPARK_EXECUTOR_EXTRA_CLASSPATH", "qux"))
+    checkOwnerReferences(executor, driverPodUid)
+  }
+
+  // There is always exactly one controller reference, and it points to the driver pod.
+  private def checkOwnerReferences(executor: Pod, driverPodUid: String): Unit = {
+    assert(executor.getMetadata.getOwnerReferences.size() == 1)
+    assert(executor.getMetadata.getOwnerReferences.get(0).getUid == driverPodUid)
+    assert(executor.getMetadata.getOwnerReferences.get(0).getController == true)
+  }
+
+  // Check that the expected environment variables are present.
+  private def checkEnv(executor: Pod, additionalEnvVars: Set[String]): Unit = {
```

Comment:
Do we not also want to be checking the values of the environment variables we are setting, not just the keys? Hence the `additionalEnvVars` should be a `Map`, not a `Set`.